src/components/select/select.ts

```diff
@@ -596,13 +596,7 @@ export default class WaSelect extends WebAwesomeFormAssociatedElement {
 
     // If no direct option was provided, find the option from the event path
     if (!option) {
-      const path = event.composedPath();
-      const tagElement = path.find(
-        el =>
-          el instanceof HTMLElement &&
-          (el.tagName.toLowerCase() === 'wa-tag' ||
-            (el instanceof HTMLElement && el.hasAttribute('part') && el.getAttribute('part')?.includes('tag'))),
-      );
+      const tagElement = (event.target as Element).closest('wa-tag[part~=tag]');
```

Comment:
I can't remember, is there any chance `event.target` can be a text node here? If so, we may want to do:

```suggestion
let targetElement = event.target.closest ? event.target : event.target.parentElement;
const tagElement = targetElement.closest('wa-tag[part~=tag]');
```